### PR TITLE
Alex grid

### PR DESCRIPTION
**Pull request by alexsielicki**
Adding option to display grid behind scatterplot in PS.

### CODE REVIEW

File: web-server/plugins/slycat-parameter-image/js/ui.js [hunk 1]
```diff
@@ -1943,7 +1959,7 @@ $(document).ready(function () {
       // and we don't want long calls to come back now and filter anything.
       if (filterxhr) {
         filterxhr.abort();
-        console.debug("filter xhr aborted because last filter was closed");
+        // console.debug("filter xhr aborted because last filter was closed");
```
Comment (Mletter1) [inline web-server/plugins/slycat-parameter-image/js/ui.js:1962]: delete

**Review by Mletter1 — CHANGES_REQUESTED** [review]

File: web-server/js/slycat-color-maps.ts [hunk 2]
```diff
@@ -64,6 +65,7 @@ export default {
       background: d3.rgb(255, 255, 255),
       null_color: "gray",
       outofdomain_color: "black",
+      scatterplot_grid_color: "rgb(80% 80% 80%)",
```
[the remark this answers is not in the document]
Comment (alexsielicki) [reply web-server/js/slycat-color-maps.ts]: Great suggestion. I created constants for all reused colors and standardized syntax to latest css version. See 2e5f12f64452156d2188c8f2a39feb8744e60137

File: web-server/plugins/slycat-parameter-image/js/ui.js (hunk 1 above)
Comment (alexsielicki) [reply web-server/plugins/slycat-parameter-image/js/ui.js]: Done in ec52d4f99280f8fd20af79b15c75b405a7fc2f9a

File: web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js [hunk 3]
```diff
@@ -1330,8 +1319,8 @@ $.widget("parameter_image.scatterplot", {
       let x_axis_width = width - self.options.margin_left - self.options.margin_right;
       // let x_remaining_width = self.svg.attr("width") - x_axis_width;
```
[the remark this answers is not in the document]
Comment (alexsielicki) [reply web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js]: Done in ec52d4f99280f8fd20af79b15c75b405a7fc2f9a

File: web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js [hunk 4]
```diff
@@ -1330,8 +1319,8 @@ $.widget("parameter_image.scatterplot", {
...
       // let x = x_remaining_width / 2 + x_axis_width + 40;
```
[the remark this answers is not in the document]
Comment (alexsielicki) [reply web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js]: Done in ec52d4f99280f8fd20af79b15c75b405a7fc2f9a

File: web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js [hunk 5]
```diff
@@ -1330,8 +1319,8 @@ $.widget("parameter_image.scatterplot", {
...
-      let x = self.options.margin_left + x_axis_width - xoffset + 40;
-      console.debug(`self.updates.update_x_label`);
+      let x = self.options.margin_left + x_axis_width + 40;
+      // console.debug(`self.updates.update_x_label`);
```
[the remark this answers is not in the document]
Comment (alexsielicki) [reply web-server/plugins/slycat-parameter-image/js/parameter-image-scatterplot.js]: Done in ec52d4f99280f8fd20af79b15c75b405a7fc2f9a

File: web-server/plugins/slycat-parameter-image/js/Components/ScatterplotGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const y_scale = useSelector(selectYScale);
+  const x_ticks = useSelector(selectXTicks);
+  const y_ticks = useSelector(selectYTicks);
+
+  const scatterplot_grid_color = slycat_color_maps.get_scatterplot_grid_color(colormap);
+
+  // Only execute the useEffect hook if show_grid is true
+  useEffect(() => {
+    if (show_grid) {
+      // console.debug("ScatterplotGrid.componentDidMount() updateGrid()");
```
[the remark this answers is not in the document]
Comment (alexsielicki) [reply web-server/plugins/slycat-parameter-image/js/Components/ScatterplotGrid.tsx]: Done in ec52d4f99280f8fd20af79b15c75b405a7fc2f9a

**Comment by alexsielicki** [comment]
Thanks for the suggestions @Mletter1 

I implemented all of them. Let me know if you find anything else.

**Comment by Mletter1** [comment]
np, I think thats it I'm gonna merge it

**Review by Mletter1 — APPROVED** [review]

**Pull request merged**